Gems/EMotionFX/Code/Source/Editor/Plugins/SkeletonOutliner/JointPropertyWidget.cpp

```diff
@@ -111,4 +154,185 @@ namespace EMotionFX
+        }
+    }
+
+    void JointPropertyWidget::OnAddToRagdoll()
+    {
+        AZ::Outcome<const QModelIndexList&> indicesOutcome;
+        SkeletonOutlinerRequestBus::BroadcastResult(indicesOutcome, &SkeletonOutlinerRequests::GetSelectedRowIndices);
+        if (!indicesOutcome.IsSuccess())
+        {
+            return;
+        }
+        auto indices = indicesOutcome.GetValue();
+
+        const AZStd::string groupName = AZStd::string::format("Add joint%s to ragdoll",
+                indices.size() > 1 ? "s" : "");
+        MCore::CommandGroup commandGroup(groupName);
+
+
+        AZStd::vector<AZStd::string> jointNames;
+        jointNames.reserve(indices.size());
+
+        // All the actor pointers should be the same
+        const uint32 actorId = indices[0].data(SkeletonModel::ROLE_ACTOR_POINTER).value<Actor*>()->GetID();
+
+        for (const QModelIndex& selectedIndex : indices)
+        {
+            const Node* joint = selectedIndex.data(SkeletonModel::ROLE_POINTER).value<Node*>();
+
+            jointNames.emplace_back(joint->GetNameString());
+        }
+
+        CommandRagdollHelpers::AddJointsToRagdoll(actorId, jointNames, &commandGroup);
+
+        AZStd::string result;
+        if (!CommandSystem::GetCommandManager()->ExecuteCommandGroup(commandGroup, result))
+        {
+            AZ_Error("EMotionFX", false, result.c_str());
+        }
+//        // All the actor pointers should be the same
+//        const uint32 actorId = modelIndices[0].data(SkeletonModel::ROLE_ACTOR_POINTER).value<Actor*>()->GetID();
+
+//        for (const QModelIndex& selectedIndex : modelIndices)
+//        {
+//            const Node* joint = selectedIndex.data(SkeletonModel::ROLE_POINTER).value<Node*>();
+
+//            jointNames.emplace_back(joint->GetNameString());
+//        }
+//        CommandRagdollHelpers::AddJointsToRagdoll(actorId, jointNames, &commandGroup);
+
+//        AZStd::string result;
+//        if (!CommandSystem::GetCommandManager()->ExecuteCommandGroup(commandGroup, result))
+//        {
+//            AZ_Error("EMotionFX", false, result.c_str());
+//        }
+    }
+
+    void JointPropertyWidget::OnSearchTextChanged()
+    {
+        m_filterString = m_filterEntityBox->text().toLatin1().data();
+
+        m_clothJointWidget->SetFilterString(m_filterString);
+        m_hitDetectionJointWidget->SetFilterString(m_filterString);
+        m_ragdollJointWidget->SetFilterString(m_filterString);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    //
+    //    AddCollidersButton
+    //
+    AddCollidersButton::AddCollidersButton(QWidget *parent)
+        :QPushButton(parent)
+    {
+        setText("Add Collider");
+        connect(this, &QPushButton::clicked, this, &AddCollidersButton::OnCreateContextMenu);
+    }
+
+    // TODO subclass ItemModel
+    enum ItemRoles
```

Comment:
suggestion: Could we use `enum class` here instead if these aren't meant to be used as flags?